src/model.ts

```diff
@@ -224,12 +224,17 @@ export class Model<T extends Record<string, ModelColumn>> {
 	 * @returns Returns the first row that matches the where clause, or null if no rows match.
 	 */
 	public async First(
-		options: Pick<GenerateQueryOptions<InferFromColumns<T>>, "where">
+		options: Pick<
+			GenerateQueryOptions<InferFromColumns<T>>,
+			"where" | "columns"
+		>
 	): Promise<InferFromColumns<T> | null> {
```

Comment:
One thing I've realised is that when the user is specifying `columns`, the return type of this method is going to be wrong - we'd only want to return the type for the columns which are being returned

There's not any particularly nice way to handle this though 🤔